pkg/resources/credentials.go

```diff
@@ -57,6 +58,14 @@ type AzureCredentials struct {
 	ClientSecret   string
 }
 
+type BaremetalCredentials struct {
+	Tinkerbell TinkerbellCredentials
+}
+type TinkerbellCredentials struct {
+	// Admin kubeconfig for Tinkerbell cluster
+	KubeConfig string
```

Comment:
Please decide if you want to name the new fields "Kubeconfig" (small c) or "KubeConfig" (capital C). Lowercase c is more common IMHO.